DMTN-165.tex

```diff
@@ -50,7 +56,289 @@
+
+In the current baseline, Rubin Observatory runs an Alert Filtering Service (AFS) within the Data Facility that allows a limited number of users ($\geq$100 simultaneous\dmreq{0343}\reqparam{numBrokerUsers}) with Data Rights to upload filters and have a limited number of alerts ($\leq$ 20 per visit\reqparam{numBrokerAlerts}) forwarded to them.
+In addition to the limited capacity, the AFS restricts the user filters to only operate on the contents of the alert packets themselves.
+There is currently no latency requirement for alert delivery through the AFS.
+
+We suggest that the lightweight alert packet design would obviate the scientific need for the AFS, reducing Project scope.
+Instead of providing filter code to run on Data Facility services, users with data rights could run filters on their own hardware with a great deal more flexibility.
+In particular, users would be free to build filters that crossmatched to external catalogs or realtime streams or performed computationally-intensive fitting or image-processing tasks.
+They would be free to choose their own programming language, libraries, and environment.
+While the need to implement their own local service would impose additional difficulty relative to a Project-provided AFS, the Project could provide simple tutorial material for a basic implementation.
+
+%\textbf{estimate data rates for 100x20.  20 full-sized alerts per filter had to come out anyway}
+
+One particular advantage of this scheme to users is that within their overall rate limit budget, they could receive more alerts from single visits at cost of higher latency.
+The AFS is likely to have had hard limits at 20 full-sized alerts per visit, after which no more would be forwarded.
+In contrast, a user wanting many more alerts from a high-priority single visit (say, a Deep Drilling field) could simply request more full-sized alerts over a longer period of time.
+
+Descoping the AFS would reduce on-Project construction and operations effort.
+
+\section{Implications for the Alert Database} \label{sec:alertdb}
+
+The store of full-sized alerts is conceptually quite similar to the ``Alert Database,'' a component of the alert system that is required as a record of all transmitted alerts but that has relatively few other enumerated requirements\dmreq{0094}.
+Particularly if flexible identifiers pointing to the full-sized alerts are used, it seems useful to investigate whether the Alert Database and the store of full-sized alerts could be the same system, technically.
+This would provide a seamless transition for the user between active and archival alerts.
+Users could retrieve full-sized alerts by their URIs, and it would be straightforward to implement additional queries (cone search, time window, etc.) by querying the Prompt Products Database and retrieving the appropriate alert URIs.
+It would also be necessary to archive the lightweight alerts, but due to their small size this should not be a large perturbation.
```

Comment:
If the lightweight alerts are a strict subset of the full-sized, they could be regenerated dynamically if needed.

Reply:
@leannep There's an arrow coming out of "Alert Packaging" to the Alert Database that I think can count for archiving the lightweight alerts.